src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_DEFAULT_LENGTH = 10;
+    private T[] array;
+    private int currentSize;
+
+    public ArrayList() {
+        array = (T[]) new Object[ARRAY_DEFAULT_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        lengthCheck(currentSize);
+        array[currentSize] = value;
+        currentSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        try {
+            lengthCheck(currentSize);
+            T[] arrayTail = Arrays.copyOfRange(array, index, currentSize);
+            array[index] = value;
+            System.arraycopy(arrayTail, 0, array, index + 1, arrayTail.length);
+            currentSize++;
+        } catch (RuntimeException e) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Can't add element. Index " + index + " is out of bounds"
+                            + " for the list size " + currentSize);
+        }
```

Comment:
bad practice. instead, validate that index is valid at the beginning of the method